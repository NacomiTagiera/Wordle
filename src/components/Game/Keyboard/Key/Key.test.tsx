import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';

import { Key } from '.';

describe('Key', () => {
  it('renders with default props', () => {
    const { asFragment } = render(<Key onClick={() => {}}>A</Key>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays provided letter', () => {
    render(<Key onClick={() => {}}>A</Key>);

    screen.getByText('A');
    expect(screen.getByTestId('key')).toHaveTextContent('A');
  });

  it('injects optional props when they are specified', () => {
    render(
      <Key onClick={() => {}} ariaLabel='A' dataKey='A' letterState='present'>
        A
      </Key>
    );

    expect(screen.getByTestId('key')).toHaveAttribute('aria-label', 'A');
    expect(screen.getByTestId('key')).toHaveAttribute('data-key', 'A');
    expect(screen.getByTestId('key')).toHaveAttribute('data-state', 'present');
  });

  it('assigns proper class name when large prop is true', () => {
    render(
      <Key onClick={() => {}} large>
        A
      </Key>
    );
    expect(screen.getByTestId('key')).toHaveClass('key--large');
  });

  it('invokes onClick function when clicked', async () => {
    const onClick = jest.fn();
    const user = userEvent.setup();
    render(<Key onClick={onClick}>A</Key>);

    await user.click(screen.getByTestId('key'));
    expect(onClick).toHaveBeenCalled();
  });
});
