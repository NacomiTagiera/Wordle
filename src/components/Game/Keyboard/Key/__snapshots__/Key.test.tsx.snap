// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key renders with default props 1`] = `
<DocumentFragment>
  <button
    class="key "
    data-key="A"
    data-testid="key"
    type="button"
  >
    A
  </button>
</DocumentFragment>
`;
